src/adapter/search-response-adapter/total-hits-adapter.ts

```diff
@@ -0,0 +1,18 @@
+import type { MeiliSearchResponse } from '../../types'
+
+export function adaptTotalHits(
+  searchResponse: MeiliSearchResponse<Record<string, any>>
+): number {
+  const {
+    hitsPerPage = 0,
+    totalPages = 0,
+    estimatedTotalHits = undefined,
+    totalHits = undefined,
+  } = searchResponse
+  if (estimatedTotalHits !== undefined) {
+    return estimatedTotalHits
+  } else if (totalHits !== undefined) {
+    return totalHits
+  }
+  return hitsPerPage * totalPages
```

Comment:
According to our internal conversation, this will change in the future, right @bidoubiwa ?

Reply:
> According to our internal conversation, this will change in the future,

What do you mean?

Reply:
The idea here is that the core team provides a real number of hits we have. Because even we doing a math here we cannot be 100% sure since we may have pages without the whole limit.

Reply:
Its the if just above 

```
else if (totalHits !== undefined) {
    return totalHits
  }
```

Thats the hit Meilisearch returns. The return is a safe guard in case total hits is missing but it should indeed not happen so I could as well remove it